containers/podman
# Shell completion

Add shell completion with cobra
 
Allow automatic generation for shell completion scripts
with the internal cobra functions (requires v1.0.0)
  
This should replace the handwritten completion scripts
and even adds support for fish and powershell(if needed?)

We can now create the scripts with
- podman completion bash
- podman completion zsh
- podman completion fish
- podman completion powershell

to test the completion run:
`source <(podman completion bash)`

I added the main completion for all commands
but this still could be improved with
better custom completion functions.
see: cmd/podman/completion/completion.go

Flag completion is for now only added to
podman generate systemd --restart-policy
for now
just to show how to implement this on flags

This is still **WIP**

Closes #6440
